bundlewrap/repo.py

```diff
@@ -513,32 +515,51 @@ def populate_from_path(self, path):
         # check requirements.txt
         try:
             with open(join(path, FILENAME_REQUIREMENTS)) as f:
-                lines = f.readlines()
+                lines = [line.strip() for line in f.readlines()]
+
+            # Ignore empty lines and comments.
+            lines = [line for line in lines if line and not line.startswith('#')]
+
+            # "-e some/editable" and "-r other_requirements.txt" are not
+            # supported.
+            lines = [line for line in lines if not line.startswith('-')]
+
+            reqs = [Requirement(line) for line in lines]
         except Exception:
             pass
         else:
-            try:
-                require(lines)
-            except DistributionNotFound as exc:
-                raise MissingRepoDependency(_(
-                    "{x} Python package '{pkg}' is listed in {filename}, but wasn't found. "
-                    "You probably have to install it with `pip install {pkg}`."
-                ).format(
-                    filename=FILENAME_REQUIREMENTS,
-                    pkg=exc.req,
-                    x=red("!"),
-                ))
-            except VersionConflict as exc:
-                raise MissingRepoDependency(_(
-                    "{x} Python package '{required}' is listed in {filename}, "
-                    "but only '{existing}' was found. "
-                    "You probably have to upgrade it with `pip install {required}`."
-                ).format(
-                    existing=exc.dist,
-                    filename=FILENAME_REQUIREMENTS,
-                    required=exc.req,
-                    x=red("!"),
-                ))
+            for req in reqs:
+                # Ignore packages with a marker that does *not* apply to
+                # the current system.
+                if req.marker is not None and not req.marker.evaluate():
+                    continue
+
+                for installed in metadata.distributions():
+                    if req.name.lower() == installed.name.lower():
+                        if req.specifier.contains(installed.version):
+                            # We're good.
+                            break
+                        else:
+                            raise MissingRepoDependency(_(
```

Comment:
In a follow-up PR we should improve this to show all missing dependencies at once, not just the first one that failed, for better UX.